internal/pkg/agent/application/application.go

```diff
@@ -296,6 +308,86 @@ func New(
 	return coord, configMgr, varsManager, nil
 }
 
+// normalizeInstallDescriptorAtStartup will check the install descriptor checking:
+// - if we just rolled back: the update marker is checked and in case of rollback we clean up the entry about the failed upgraded install
+// - check all the entries:
+//   - verify that the home directory for that install still exists (remove what does not exist anymore)
+//   - TODO check TTLs of installs to schedule delayed cleanup while the agent is running
+//
+// This function will NOT error out, it will log any errors it encounters as warnings but any error must be treated as non-fatal
+func normalizeInstallDescriptorAtStartup(log *logger.Logger, topDir string, now time.Time, initialUpdateMarker *upgrade.UpdateMarker, rollbackSource rollbacksSource) {
+	// Check if we rolled back and update the install descriptor
```

Comment:
For example, this comment references an install descriptor that no longer exists in the code.

Reply:
There are a couple more references to install descriptors in logs and such as well.

Reply:
cleaned up in https://github.com/elastic/elastic-agent/pull/9643/commits/dcf0ede92e3ba49bd079e551f66a24f91201fc79